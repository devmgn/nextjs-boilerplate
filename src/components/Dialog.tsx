'use client';

import styled from 'styled-components';

const Dialog = styled.div`
  width: min(100%, 400px);
  max-height: 100%;
  padding: 24px;
  overflow: auto;
  background-color: #fff;
  border-radius: 8px;
`;

export default Dialog;
