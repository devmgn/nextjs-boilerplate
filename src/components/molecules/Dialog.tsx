import { m } from 'framer-motion';
import styled from 'styled-components';
import { TRANSITION } from '@/constants';

const StyledDialog = styled(m.div)`
  width: min(300px, 100%);
  max-height: fill-available;
  overflow: auto;
  background-color: #fff;
  -webkit-overflow-scrolling: touch;
`;

type DialogProps = {
  children: React.ReactNode;
};

const Dialog: React.FC<DialogProps> = ({ children }) => {
  return (
    <StyledDialog
      key="modal-backdrop"
      initial="inactive"
      animate="active"
      exit="inactive"
      variants={{
        active: { y: 0 },
        inactive: { y: '100vh' },
      }}
      transition={TRANSITION}
      data-scroll-lock-scrollable
    >
      <button type="button" data-modal-deactivate>
        閉じる
      </button>
      {children}
    </StyledDialog>
  );
};

export default Dialog;
