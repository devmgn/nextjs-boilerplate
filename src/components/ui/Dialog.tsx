import styled from 'styled-components';

const StyledDialog = styled.div`
  width: min(300px, 100%);
  max-height: calc(100vh - 32px);
  padding: 16px;
  background-color: #fff;
  overflow: auto;
  -webkit-overflow-scrolling: touch;
  @supports (-webkit-touch-callout: none) {
    max-height: calc(100dvh - 32px);
  }
`;

type DialogProps = {
  children: React.ReactNode;
};

const Dialog: React.FC<DialogProps> = ({ children }) => {
  return (
    <StyledDialog data-scroll-lock-scrollable>
      <button type="button" data-modal-deactivate>
        閉じる
      </button>
      {children}
    </StyledDialog>
  );
};

export default Dialog;
