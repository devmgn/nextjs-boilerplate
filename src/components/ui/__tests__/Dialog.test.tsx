import { render } from '@testing-library/react';
import Dialog from '../Dialog';

describe('Dialog', () => {
  test('スナップショットテスト', () => {
    const { asFragment } = render(<Dialog>children</Dialog>);
    expect(asFragment()).toMatchSnapshot();
  });
});
