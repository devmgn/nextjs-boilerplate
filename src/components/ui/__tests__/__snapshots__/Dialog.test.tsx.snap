// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog スナップショットテスト 1`] = `
<DocumentFragment>
  .c0 {
  width: min(300px,100%);
  max-height: calc(100vh - 32px);
  padding: 16px;
  overflow: auto;
  background-color: #fff;
  -webkit-overflow-scrolling: touch;
}

@supports (-webkit-touch-callout:none) {
  .c0 {
    max-height: calc(100dvh - 32px);
  }
}

<div
    class="c0"
    data-scroll-lock-scrollable="true"
  >
    <button
      data-modal-deactivate="true"
      type="button"
    >
      閉じる
    </button>
    children
  </div>
</DocumentFragment>
`;
